adapters/common/src/nemo_fabric_adapters/common/lifecycle.py

```
@@ -614,7 +631,36 @@ async def _handle_invoke_openai_stream(
+    try:
+        request = AgentRunRequest.from_mapping(payload.get("request"))
+        context = RuntimeContext.from_mapping(payload.get("runtime_context"))
+    except Exception as error:
+        raise LifecycleError(
+            "lifecycle_invalid_request",
+            "Invocation does not match the typed adapter contract",
+        ) from error
+    return request, context
+
+
+def _typed_result(result: Any) -> dict[str, Any]:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `result`

(ANN401)